pasqal_cloud/__init__.py

```diff
@@ -234,44 +236,115 @@ def cancel_batch(self, id: str) -> Batch:
 
     def rebatch(
         self,
-        id: str,
-        job_ids: Union[List[str], EmptyFilter] = EmptyFilter(),
-        status: Union[List[str], EmptyFilter] = EmptyFilter(),
-        min_runs: Union[int, EmptyFilter] = EmptyFilter(),
-        max_runs: Union[int, EmptyFilter] = EmptyFilter(),
-        start_date: Union[datetime, EmptyFilter] = EmptyFilter(),
-        end_date: Union[datetime, EmptyFilter] = EmptyFilter(),
+        id: Union[UUID, str],
+        filters: Optional[RebatchFilters] = None,
     ) -> Batch:
         """
         Retry a list of jobs matching filters in a new batch.
 
         Args:
-            id: the id of the batch to rebatch jobs from.
-            job_ids: list of job ids to retry.
-            status: status or list of statuses.
-                Will retry jobs currently matching this/these status/es.
-            min_runs: retry jobs with more or as many as this amount of runs.
-            mas_runs: retry jobs with less or as many as this amount of runs.
-            start_date: retry jobs created at or after this datetime.
-            end_date: retry jobs created at or before this datetime.
+            id: id of the batch to re-create
+            filters: filters to be applied to find the jobs that will be retried
+
+        Returns:
+            Batch: the new re-created batch
 
         Raises:
             RebatchError if rebatch call failed.
         """
+        if filters is None:
+            filters = RebatchFilters()
+        elif not isinstance(filters, RebatchFilters):
+            raise ValueError(
+                f"Filters needs to be a RebatchFilters instance, not a {type(filters)}"
+            )
+
         try:
-            new_batch_data = self._client.rebatch(
-                id,
-                job_ids=job_ids,
-                status=status,
-                min_runs=min_runs,
-                max_runs=max_runs,
-                start_date=start_date,
-                end_date=end_date,
+            new_batch_data = self._client._rebatch(
+                batch_id=id,
+                filters=filters,
             )
         except HTTPError as e:
             raise RebatchError(e) from e
         return Batch(**new_batch_data, _client=self._client)
 
+    def get_jobs(
+        self,
+        filters: Optional[JobFilters] = None,
+        from_index: int = 0,
+        page_size: int = 100,
+    ) -> dict[str, Union[int, List[Job]]]:
```

Comment:
I'd recommend using either a TypedDict or a dedicated python class to hold the paginated data to benefit from type hinting